### DISCUSSION TITLE
Is there a way to ignore updates for particular parents for auto-materialize?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17538] [Q&A] [answered]
In auto-materialization, is there a way to say “only automaterialize my downstream asset if upstream asset A is newer than my last run. If upstream asset B is newer than my last run, ignore”: <https://docs.dagster.io/concepts/assets/asset-auto-execution>

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698776416784969?thread_ts=1698776416.784969&cid=C01U954MEER).*

**Comment by OwenKephart** [comment, marked as answer]
Currently, there's not a way to do this explicitly (although this is on the radar), although in some cases (e.g. B updates more frequently than A), only materializing when [all parents have been updated](https://docs.dagster.io/concepts/assets/asset-auto-execution#auto-materialize-only-once-all-parents-have-been-updated) would do essentially the same thing.

**Reply by OwenKephart** [reply]
Created an issue to track this here: https://github.com/dagster-io/dagster/issues/17550

**Reply by achen187** [reply]
ok thanks. makes sense. Yeah, the "all parents have been updated" doesnt quite work for us. We basically dont ever want to automaterialize for B. But if theres other assets upstream (a, c), we want to automaterialize for those. Basically, we want to pass a blacklist into the asset for which upstream dependencies to automaterialize for.

Whats an alternative for this behavior if we dont use automaterializiation? sensor?

**Reply by OwenKephart** [reply]
I think in the meantime yeah an asset sensor / potentially multi-asset sensor might be the best option here
